### PR TITLE
Added Data Source Hints

### PR DESCRIPTION
**Pull request by AbdulR3hman**
*Issue #, if available:*

Allows engine to be aware of list of configurations through hints; specifically starting with Collation configuration. 

*Description of changes:*
* Introduced new Hint pushdown subtype

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java [hunk 1]
```diff
@@ -72,6 +73,16 @@ public Map.Entry<String, List<OptimizationSubType>> withSupportedSubTypes(Pushdo
             }
             return new SimpleImmutableEntry<String, List<OptimizationSubType>>(SUPPORTS_COMPLEX_EXPRESSION_PUSHDOWN.getOptimization(), Arrays.stream(subTypesList).map(pushdownSubTypes -> new OptimizationSubType(pushdownSubTypes.getSubType(), pushdownSubTypes.getProperties())).collect(Collectors.toList()));
         }
+    },
+    DATA_SOURCE_HINTS("data_source_hints")
```
Comment (chngpe) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java:77]: Can you please add a comment of purpose on this parent type

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java [hunk 2]
```diff
@@ -72,6 +73,16 @@ public Map.Entry<String, List<OptimizationSubType>> withSupportedSubTypes(Pushdo
...
+    {
+        public Map.Entry<String, List<OptimizationSubType>> withSupportedSubTypes(PushdownSubTypes... subTypesList)
+        {
+            if (!Arrays.stream(subTypesList).allMatch(pushdownSubTypes -> pushdownSubTypes instanceof HintsSubtype)) {
+                throw new IllegalArgumentException("Data Source Hunts  must contain valid data source hint subtypes.");
```
Comment (chngpe) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java:82]: nit: typo

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java (hunk 1 above)
Comment (AbdulR3hman) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java:79]: done;

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java (hunk 2 above)
Comment (AbdulR3hman) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/DataSourceOptimizations.java]: done

File: athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/pushdown/HintsSubtype.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connector.lambda.metadata.optimizations.pushdown;
+
+public enum HintsSubtype
+    implements PushdownSubTypes
+{
+    //This indicates that the connector does not support the same default collate setting of Athena
+    NON_DEFAULT_COLLATE("non_default_collate");
```
Comment (AbdulR3hman) [inline athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/pushdown/HintsSubtype.java:26]: This basically indicate that this connector is non-default; meaning it doesn't match Athena's engine default sorting

Comment (AbdulR3hman) [reply athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/metadata/optimizations/pushdown/HintsSubtype.java:26]: alternative name; `UNMATCHED_COLLATE` or `NON_MATHCING_COLLATE`

**Review by burhan94 — APPROVED** [review]

**Pull request merged**